edb/server/compiler/compiler.py

```diff
@@ -1026,12 +1026,27 @@ def _compile_ql_migration(
                         mstate.target_schema,
                         guided_diff,
                     )
-
                     proposed_steps = []
 
                     if proposed_ddl:
-                        for ddl_text, top_op in proposed_ddl:
-                            prompt_id, prompt_text = top_op.get_user_prompt()
+                        for ddl_text, ast, top_op in proposed_ddl:
+                            # get_ast has a lot of logic for figuring
+                            # out when an op is implicit in a parent
+                            # op. get_user_prompt does not have any of
+                            # that sort of logic, which makes it
+                            # susceptible to producing overly broad
+                            # messages. To avoid duplicating that sort
+                            # of logic, we recreate the delta from the
+                            # AST, and extract a user prompt from
+                            # *that*.
+                            # This is stupid, and it is slow.
+                            top_op2 = s_ddl.cmd_from_ddl(
```

Comment:
Hmm, this is quite unfortunate.  We should be able to extract the elision logic from `get_ast` to a helper method and try to reuse that in `get_user_prompt` and friends.  That would be quite involved, though, so I'm fine with the hack for now.

Reply:
Yeah, that's about where I am.